plugins/examples/tables/src/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { registerResourceTableColumnsProcessor } from '@kinvolk/headlamp-plugin/lib';
+import { ActionButton } from '@kinvolk/headlamp-plugin/lib/CommonComponents';
+import Pod from '@kinvolk/headlamp-plugin/lib/K8s/pod';
+import { Menu, MenuItem, Typography } from '@material-ui/core';
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+
+// Will show a 3-dot menu with two options: Details and Delete.
+function ContextMenu(props: { pod: Pod }) {
+  const { pod } = props;
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const history = useHistory();
+
+  function openMenu(event: any) {
+    setAnchorEl(event.currentTarget);
+  }
+
+  function handleClose() {
+    setAnchorEl(null);
+  }
+
+  return (
+    <>
+      <ActionButton
+        description="Open pod context menu"
+        icon="mdi:dots-vertical"
+        onClick={openMenu}
+      />
+      <Menu anchorEl={anchorEl} open={Boolean(anchorEl)} onClose={handleClose}>
+        <MenuItem
+          onClick={() => {
+            history.push(pod.getDetailsLink());
+          }}
+        >
+          <Typography>Details</Typography>
+        </MenuItem>
+        <MenuItem
+          onClick={() => {
+            alert('Boom! Pod deleted! (just kidding)');
+            handleClose();
+          }}
+        >
+          <Typography>Delete</Typography>
+        </MenuItem>
+      </Menu>
+    </>
+  );
+}
+
+// Add a context menu to the pods table.
+// You can see it by going to the pods list view: there will be a 3-dot menu for every row.
+registerResourceTableColumnsProcessor(function setupContextMenuForPodsList({ id, columns }) {
+  // Not checking for the table ID will add the context menu to all resource tables.
+  if (id === 'headlamp-pods') {
```

Comment:
How do users of the function know which table ids are available, or how to find them?

In another processor there's an object with the default ids they can import. Maybe tell them in the docs for id where to look, and the convention id naming follows.

Reply:
The ids do get automatically generated for known kinds of pods.
Keeping a list of the ids will be hard to maintain IMO, and it will not of course include any IDs for tables that plugins add, so developers will likely list all ids before selecting the ones they want to use.

But I agree that better docs are needed. Let's just add that in a different PR.

Reply:
I'd prefer if the docs were updated here.

How about this?

> How do you find the table id? They follow a convention. `headlamp-${pluralName}`. For example, Pods would be `headlamp-pods` and for Namespaces it would be `headlamp-namespaces`. Plugins can add their own table ids, so you can either look at what the plugin uses for the id.